src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -665,42 +668,47 @@ def handler(args):
     context._cli_command(name, handler=handler, argument_loader=generic_wait_arguments_loader, **kwargs)  # pylint: disable=protected-access
 
 
-def _cli_generic_show_command(context, name, getter_op, **kwargs):
+def _cli_generic_show_command(context, name, getter_op, custom_command=False, **kwargs):
```

Comment:
Yes, make this accept `source_kwarg` (no default) and have show_command pass 'command_type' and custom_show_command pass 'custom_command_type'.  It will simplify your code. Ditto for the private wait command.

Reply:
The problem with passing along `source_kwarg` is that it does not give any indication what all the possible choices are for it. 
[Validation](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/commands/__init__.py#L825-L827) also needs to be done on that `source_kwarg` in the case that a spelling typo is made unless we are relying on this being caught elsewhere (not ideal).

Using a flag gives the user clear information that there are to possible choices and leaves less room for error.
To lessen the `<>if custom_command else <>` statements, I've added a util [method](https://github.com/Azure/azure-cli/pull/6740/files#diff-5241dc629edabeefd74a7984150494bbR315) that selects the right `source_kwarg` that will be utilized in all the cases.